extensions/2.0/Khronos/KHR_materials_transmission/README.md

```diff
@@ -0,0 +1,126 @@
+
+*f* = mix(**f**<sub>*diffuse*</sub>, **L**<sub>transmitted</sub> * `baseColor`, **T**) + *f*<sub>*specular*</sub>
+
+Where **L**<sub>transmitted</sub> is the colour of light coming through the material, towards the eye. It is tinted (multiplied) by the `baseColor`, and already has refraction taken into account (as discussed below). *T* is the transmission value defined by this extension.
+
+## A Note about Blend Mode
+
+Although the math here is straightforward, it doesn't consist of a single blend equation and so, in practice, can't be rendered in a single draw call. Also note that alpha coverage still needs to work as it did before, blending between the destination framebuffer and this material. Therefore, `blendMode` should, in general, be set to "OPAQUE" unless alpha coverage is used.
+
+<figure>
+  <img src="./figures/TransmissionWithMask.png"/>
+<figcaption><em>Alpha coverage and optical transparency can be used at the same time so that some areas of a surface are transparent while others disappear entirely.</em></figcaption>
+</figure>
+
+## Transparent Metals
+
+The metallic parameter of a glTF material effectively scales the `baseColor` of the material toward black while, at the same time scaling the F0 (reflectivity) value towards 1.0. This effectively makes the material opaque for metallic values of 1.0 because transmitted light is attenuated out by absorption. This leaves only the specular component.
+
+## Modeling Absorption
+
+Absorption is usually defined as an amount of light at each frequency that is absorbed over a given distance through a medium (usually described by Beer’s Law). However, since this extension deals exclusively with very thin surfaces, we can treat absorption as a constant. In fact, rather than absorbed light, we can talk about its inverse: transmitted light. The `baseColor` of the material serves this purpose as it already defines how the light that penetrates the surface is colored by the material. In this model, the transmitted light will be modulated by this color as it passes through.
+
+<figure>
+  <img src="./figures/ConstantTransmission.png"/>
+<figcaption><em>The baseColor of the material is used to tint the light being transmitted.</em></figcaption>
+</figure>
+
+### Modeling Refraction
+
+Since the surface is considered to be infinitely thin, we end up with a non-physically correct model. One the one hand, the material has no volume so light shouldn't be refracted when passing through. However, it should be expected that microfacets in a rough surface would refract the transmitted light in such a way as to blur the transmitted image. `How do we define this, mathematically??? By its nature, it's non-physically-based.`
```

Comment:
Do you consider the effect of the index of refraction on thin refraction? E.g., as the index of refraction approaches 1, does the blur effect disappear?

Reply:
Yes, I think IOR should affect the amount of blur but I wasn't sure how to define anything mathematically yet. How much does an infinitely-thin surface refract light? Well, none...

There would be a (1-F) affect from light hitting the micro-facets, I guess, but no refraction.

Unless I'm not thinking about this correctly.

Reply:
Am I right in assuming that we do want a refractive blurring effect, even with an infinitely thin surface?
What do offline renderers do in this case? That should be our target reference.

For real-time, we can provide a recommended implementation but I think we need a higher-quality target so that real-time renderers are free to set their sights higher.

Reply:
> How much does an infinitely-thin surface refract light?

If I am not mistaken, it depends on the orientation of the microfacets on the two sides of the object enclosing the infinitesimal small volume. If the facets on both sides are aligned such that the light ray entering the thin "volume" exists through a surface that has exactly the same orientation, there is no refraction. On the other hand, if the orientation is different, refraction will occur (if IOR != 1). This can be approximated with a microfacet lobe that is centered around the reflection direction flipped to the lower hemisphere wrt. the macro-surface normal. Here is an example how it is implemented in MDL for offline rendering: [libbsdf, line 514](https://github.com/NVIDIA/MDL-SDK/blob/2019.1.4/src/mdl/jit/libbsdf/libbsdf.cpp#L514). Another good source regarding offline rendering is the [Thin Surface BSDF from Imageworks (slide 34)](https://blog.selfshadow.com/publications/s2017-shading-course/imageworks/s2017_pbs_imageworks_slides_v2.pdf). They also show comparisons between actual modelled thin surfaces and their thin BTDF, and derive a roughness mapping that includes the IOR.

Reply:
Thank you. I've modified the spec to reflect this.